book-2-mvc/chapters/ADD_AND_UPDATE_DATA_IN_MVC.md

```diff
@@ -0,0 +1,269 @@
+# Adding and Updating Data with MVC
+
+In this chapter you'll continue to implement CRUD for the DogGo application by adding Create, Edit, and Delete routes for our dog owners.
```

Comment:
Why have we switched from Walters to Owners? Looks like you've got a a mix of references to both Walters and Owners throughout this chapter. I think we should stick with Walkers unless there's a reason not to.

Reply:
I didn't think the app would necessarily include functionality to add walkers. They'd just come pre-populate. I thought it would be more interesting to start with them since they had images...

As for switching to owners, my thinking was this: The assignment after the lesson would be to implement CRUD for Dogs. The next day when we talk about Auth, we can say that the owner Create page is really just a Register. And when accessing the Edit route, users/owners should only be able to edit their own data, and then when viewing Walkers, they'd only see walkers that were in their neighborhood. The assignment from there for students would be to update the Dog controllers so that users would no longer have to put in an Owner Id when creating a Dog. Instead the Dog would be owned by whoever the logged in user is. Also, when viewing the Dogs page, a user would only see their own dogs and could only edit their own dogs

Reply:
I think we need to talk more about this. While I'm a big believer in making our exercises as real as we can, I'm concerned that this approach might be favoring realism at the expense of over-burdening the students while they are trying to learn MVC.